### ISSUE TITLE
We should have a tool that outputs code that can be compiled

### ISSUE THREAD

**Issue by bradgrantham-lunarg** [LunarG/gfxreconstruct#572]
(Deferred from #158)

A switch-controlled option to output compilable (or near-compilable) source would also be helpful.

**Comment by kinochi** [comment]
Hi currently I see toascii tool only outputs the function name , it would be useful if it could output the parameters as well passed to it just like vkapi layer.

**Comment by bradgrantham-lunarg** [comment]
@kinochi, the format is in flux at the moment, but we're moving to JSON output in `toascii`.  You can take a look at `dev` branch to see a much-improved `toascii` but I think the JSON format will change before it lands in the SDK.

**Comment by kinochi** [comment]
Thanks , I didn't ran the latest version of tool , it seems to give much of the information I need ! I really appreciate the quick reply and I admire your work.

Hope we'll be able to get the tools you mentioned too . Would love to have a compilable sample straight out of a gfxr.

**Comment by andrew-lunarg** [comment]
We have multi-gigabyte captures. Are we thinking of outputting gigabytes of code with all the large blobs of data from the trace in static variables in the code?
I think we would need to define how much generated code is reasonable to build and what use cases would fit within those constraints before we start this.
Clearly, a frame or two of vkcube is feasible and chunk of gameplay from a AAA title is not, so what interesting uses are left in the middle?

**Comment by per-mathisen-arm** [comment]
I think the most reasonable way to do this would be to write the blobs of data out as separate files to disk. I suspect one of the main uses of this would be to extract such resources and use them in small programs reproducing a very small portion of the original trace, eg to try to trigger and minimally reproduce a particular bug or bottleneck.

**Comment by bradgrantham-lunarg** [comment]
@andrew-lunarg wrote:
> We have multi-gigabyte captures. Are we thinking of outputting gigabytes of code with all the large blobs of data from the trace in static variables in the code?

@per-mathisen-arm wrote:
> I think the most reasonable way to do this would be to write the blobs of data out as separate files to disk.

This is how we've noted we would store complete `toascii` dumps to disk, by referencing blobs as separate files.  I think compileable code would be the same.
